src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/MandateForChange.kt

```diff
@@ -13,27 +13,29 @@ class MandateForChange(
 ) {
   fun hasNoMandate(crn: String, convictions: Collection<Conviction>): Boolean =
     convictions
-      .filter { it.sentence?.terminationDate == null }
-      .let { activeConvictions ->
-        activeConvictions.none {
-          it.sentence != null && (isCustodial(it.sentence) || hasNonRestrictiveRequirements(crn, it.convictionId))
-        }
+      .filter { currentSentence(it.sentence) }
+      .none {
+        isCustodial(it.sentence) || hasNonRestrictiveRequirements(crn, it.convictionId)
       }.also { log.debug("Has no mandate for change: $it") }
 
-  private fun isCustodial(sentence: Sentence) =
-    sentence.sentenceType.code in custodialSentences
+  private fun currentSentence(sentence: Sentence?) =
+    sentence != null && sentence.terminationDate == null
+
+  private fun isCustodial(sentence: Sentence?) =
+    sentence?.sentenceType?.code in custodialSentences
 
   private fun hasNonRestrictiveRequirements(crn: String, convictionId: Long): Boolean =
     communityApiClient.getRequirements(crn, convictionId)
       .filter { excludeUnpaidWork(it) }
       .any { it.restrictive != true }
       .also { log.debug("Has non-restrictive requirements: $it") }
 
-  private fun excludeUnpaidWork(it: Requirement) =
-    it.requirementTypeMainCategory?.code != "W"
+  private fun excludeUnpaidWork(it: Requirement): Boolean =
+    it.requirementTypeMainCategory?.code !in unpaidWorkAndOrderExtended
 
   companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
     private val custodialSentences = arrayOf("NC", "SC")
+    private val unpaidWorkAndOrderExtended = arrayOf("W", "W1")
```

Comment:
Was there a W2?

Reply:
Not on the offender provided. I have seen no evidence of W2 as yet